### PR TITLE
Adds the audit-task-generator tool.

### PR DESCRIPTION
**Pull request by dbernstein**
See README for instructions on how to use.

### CODE REVIEW

File: audit-task-generator/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+set -Daws.region or use an environment variable (AWS_REGION).
+
+The tool builds into an executable JAR file  which can be run using the following command: 
+```
+java -Daws.profile="<profile your-profile>" -jar audit-task-generator-<version>.jar
+```
+This will display help text that indicates the necessary parameters.
+
+# More Information
+Further documentation for this tool can be found [on the DuraCloud wiki](https://wiki.duraspace.org/display/DURACLOUDDOC/Auxiliary+Tools)
```
Comment (bbranan) [inline audit-task-generator/README.md]: This last section of the README should be dropped, as this is a mill tool (not in the Auxiliary set)

File: audit-task-generator/src/main/java/org/duracloud/mill/audit/taskgenerator/AuditTaskGenerator.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        final TaskQueue queue = new SQSTaskQueue( auditQueueName );
+        log.info( "Created task queue for queue " + auditQueueName );
+
+        String storeId = contentStore.getStoreId();
+        Iterator<String> contentIds = this.contentStore.getSpaceContents( this.spaceId );
+
+        while (contentIds.hasNext()) {
+            final String contentId = contentIds.next();
+            Map<String, String> props = contentStore.getContentProperties( spaceId, contentId );
+            Content content = contentStore.getContent( spaceId, contentId );
```
Comment (bbranan) [inline audit-task-generator/src/main/java/org/duracloud/mill/audit/taskgenerator/AuditTaskGenerator.java]: Is this call necessary? The content stream shouldn't be needed, and the properties have already been retrieved on the previous line.

File: audit-task-generator/src/main/java/org/duracloud/mill/audit/taskgenerator/AuditTaskGenerator.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            AuditTask task = new AuditTask();
+            task.setAction( AuditTask.ActionType.ADD_CONTENT.name() );
+            task.setUserId( this.username );
+            task.setDateTime( String.valueOf( System.currentTimeMillis() ) );
```
Comment (bbranan) [inline audit-task-generator/src/main/java/org/duracloud/mill/audit/taskgenerator/AuditTaskGenerator.java]: Should this date reflect the time this tool is run or the time the content item was added in the background (in which case, the date would come from the content properties)?

File: audit-task-generator/README.md (hunk 1 above)
Comment (dbernstein) [reply audit-task-generator/README.md]: will do.

File: audit-task-generator/src/main/java/org/duracloud/mill/audit/taskgenerator/AuditTaskGenerator.java (hunk 2 above)
Comment (dbernstein) [reply audit-task-generator/src/main/java/org/duracloud/mill/audit/taskgenerator/AuditTaskGenerator.java]: I will take a closer look and remove if it is indeed unnecessary.

File: audit-task-generator/src/main/java/org/duracloud/mill/audit/taskgenerator/AuditTaskGenerator.java (hunk 3 above)
Comment (dbernstein) [reply audit-task-generator/src/main/java/org/duracloud/mill/audit/taskgenerator/AuditTaskGenerator.java]: yes - probably the latter.  I'll fix.

**Pull request merged**